rfcs/0050-merge-bot-for-maintainers.md

```diff
@@ -3,65 +3,99 @@ feature: merge-bot-for-maintainers
+shepherd-team: @aanderse, @globin, @grahamc, @worldofpeace
+shepherd-leader: @globin
 related-issues: (will contain links to implementation PRs)
 ---
 
 # Summary
 [summary]: #summary
 
-Allow maintainers of packages and modules in Nixpkgs to submit their own changes
-when certain criteria are fullfilled.
+Allow maintainers of packages in Nixpkgs to submit their own changes
+when a trusted reviewer reviewed their change and other criteria are fullfilled.
+When the criteria are fullfilled a change can be merged using
+
+    @grahamcofborg merge
 
 # Motivation
 [motivation]: #motivation
 
-Nixpkgs is growing. More people should be able to help out with merging changes.
-Merging changes typically requires write permission, however, that gives the
-possibility for Nixpkgs-wide changes and decreases the security.
+Nixpkgs is growing. Community members want to be able and should be able to help out
+with merging changes. Merging changes typically requires write permission, however,
+that gives the possibility for Nixpkgs-wide changes and decreases the security.
 
 By allowing maintainers of packages and modules to submit their own changes the
 load on members of the @nixpkgs-committers that have write permission group is
 reduced.
 
+A new group, trusted reviewers, is introduced to help out with the reviewing
+process to improve the reviewing culture by introducing peer-review and making
+the whole process more inclusive.
+
 # Detailed design
 [design]: #detailed-design
 
-Maintainers of packages and modules will be able to submit their own changes
-when the following criteria are met:
-1. The label `11.by: package-maintainer` is set by @grahamcofborg
-2. The packages/modules are not new.
-3. All packages/modules affected need to be maintained by the maintainer.
+Community members will be able to submit their changes using a bot after a
+positive review of both the maintainers of the expressions they modify
+("relevant maintainers") as well that of a trusted reviewer.
 
-When these conditions are met the maintainer can write
+## Scope
 
-    @grahamcofborg merge
+- Only packages are considered. Modules are excluded because of how intertwined modules are.
+- Only `master` and `staging` branches are considered. In the future release branches may be included as well.
+
+## Trusted reviewers
+
+A new trusted reviewers group is introduced, called @nixpkgs-reviewers. This
+group is permission-wise positioned between @nixpkgs-maintainers and
+@nixpkgs-committers and can be considered a stepping stone towards becoming a
+member of @nixpkgs-committers and obtaining push permission.
+
+This new group is initially implemented with GitHub Teams. In the future this
+may be done differently, e.g. when narrower control lists are pursued.
+
+## Merge bot behavior
+
+The bot @grahamcofborg will support two new commands, `@grahamcofborg merge` and
+`@grahamcofborg stop` for respectively merging a PR and aborting a PR merge.
```

Comment:
I think `stop` is too generic (stop build?). Maybe `dontmerge`?

Reply:
Maybe `abort-merge` or `cancel-merge`?